src/webui/api/torrentscontroller.cpp

```diff
@@ -1041,3 +1041,70 @@ void TorrentsController::categoriesAction()
 
     setResult(categories);
 }
+
+void TorrentsController::addTagsAction()
+{
+    checkParams({"hashes", "tags"});
+
+    const QStringList hashes {params()["hashes"].split('|')};
+    const QStringList tags {params()["tags"].split(',', QString::SkipEmptyParts)};
+
+    for (const QString &tag : tags) {
+        const QString tagTrimmed {tag.trimmed()};
```

Comment:
@fusioneer-GitHub 
Please change this to:
```c++
for (const QString &tag : tags) {
    applyToTorrents(hashes, [&tag](BitTorrent::TorrentHandle *const torrent)
    {
        torrent->addTag(tag.trimmed());
    });
}
```

Reply:
@Chocobo1, why?
https://github.com/qbittorrent/qBittorrent/pull/10527#discussion_r290877347

Reply:
>@Chocobo1, why do you suggest to trim it for each iteration instead of making it only once?

I have no idea what or how.
Please give your direct suggestion to @fusioneer-GitHub about what to do.

Reply:
@Chocobo1, don't you follow this thread carefully? https://github.com/qbittorrent/qBittorrent/pull/10527#discussion_r290860805

>Please give your direct suggestion to @fusioneer-GitHub about what to do.

Either leave it as is currently or change it to not produce extra variable (as suggested in https://github.com/qbittorrent/qBittorrent/pull/10527#discussion_r290860805).

Reply:
I'm not sure what is the point here? What are we aiming to achieve? The trim happens only once for each items in `tags`, no?
IMO https://github.com/qbittorrent/qBittorrent/pull/10527#discussion_r291838277 is even better than https://github.com/qbittorrent/qBittorrent/pull/10527#discussion_r290860805 because there is less variables, isn't?

Reply:
I understand it now, just leave it as-is.